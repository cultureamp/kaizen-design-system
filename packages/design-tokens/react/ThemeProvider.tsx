import React from "react"
import { ThemeManager, defaultTheme, Theme } from "../"

export const ThemeContext = React.createContext<Theme>(defaultTheme)

/**
 * Wrap your application in this provider using a ThemeManager, to synchronise it with a react context.
 * This allows child react elements to more easily use theme variables, using the {@link useTheme} hook.
 */
export const ThemeProvider = ({
  themeManager,
  ...props
}: {
  themeManager: ThemeManager
  children: React.ReactNode
}) => {
  const [theme, setTheme] = React.useState<Theme>(
    themeManager.getCurrentTheme()
  )
  React.useEffect(() => {
    let cancelled = false
    const listener = (newTheme: Theme) => {
      if (!cancelled) setTheme(newTheme)
    }
    themeManager.addThemeChangeListener(listener)
    return () => {
      cancelled = true
      themeManager.removeThemeChangeListener(listener)
    }
  }, [])

  return (
    <ThemeContext.Provider value={theme}>
      {props.children}
    </ThemeContext.Provider>
  )
}

export const useTheme = () => React.useContext(ThemeContext)
