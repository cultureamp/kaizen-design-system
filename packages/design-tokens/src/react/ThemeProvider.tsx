import React, { useState } from "react"
import { ThemeManager, defaultTheme, Theme } from ".."

export const ThemeContext = React.createContext<Theme>(defaultTheme)

/**
 * Wrap your application in this provider using a ThemeManager, to synchronise it with a react context.
 * This allows child react elements to more easily use theme variables, using the {@link useTheme} hook.
 */
/**
 * @deprecated ThemeProvider is deprecated. Please use KaizenProvider from "@kaizen/components" instead.
 */
export const ThemeProvider = ({
  themeManager,
  ...props
}: {
  themeManager?: ThemeManager
  children: React.ReactNode
}): JSX.Element => {
  const [themeManagerInstance] = useState(
    () => themeManager || new ThemeManager(defaultTheme)
  )

  const [theme, setTheme] = React.useState<Theme>(
    themeManagerInstance.getCurrentTheme()
  )
  React.useEffect(() => {
    let cancelled = false
    const listener = (newTheme: Theme): void => {
      if (!cancelled) setTheme(newTheme)
    }
    themeManagerInstance.addThemeChangeListener(listener)
    return () => {
      cancelled = true
      themeManagerInstance.removeThemeChangeListener(listener)
    }
  }, [])

  return (
    <>
      <ThemeContext.Provider value={theme}>
        {props.children}
      </ThemeContext.Provider>
      <link
        rel="preload"
        href="https://d1e7r7b0lb8p4d.cloudfront.net/fonts/inter/inter-bold.woff2"
        as="font"
        type="font/woff2"
        crossOrigin="anonymous"
      />
      <link
        rel="preload"
        href="https://d1e7r7b0lb8p4d.cloudfront.net/fonts/inter/inter-demi-bold.woff2"
        as="font"
        type="font/woff2"
        crossOrigin="anonymous"
      />
      <link
        rel="preload"
        href="https://d1e7r7b0lb8p4d.cloudfront.net/fonts/inter/inter-medium.woff2"
        as="font"
        type="font/woff2"
        crossOrigin="anonymous"
      />
      <link
        rel="preload"
        href="https://d1e7r7b0lb8p4d.cloudfront.net/fonts/inter/inter-regular.woff2"
        as="font"
        type="font/woff2"
        crossOrigin="anonymous"
      />
      <link
        rel="preload"
        href="https://d1e7r7b0lb8p4d.cloudfront.net/fonts/tiempos/tiempos-headline-bold.woff2"
        as="font"
        type="font/woff2"
        crossOrigin="anonymous"
      />
    </>
  )
}

export const useTheme = (): Theme => React.useContext(ThemeContext)
