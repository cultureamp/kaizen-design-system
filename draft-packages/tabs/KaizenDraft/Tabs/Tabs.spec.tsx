import { cleanup, fireEvent, render } from "@testing-library/react"
import * as React from "react"

import { Tabs } from "."

const styles = require("./styles.scss")

afterEach(cleanup)

describe("Tabs", () => {
  it("renders basic tabs", () => {
    const tabs = [{ label: "One" }, { label: "Two" }]
    const { container } = render(<Tabs tabs={tabs} />)

    expect(container.firstChild).toMatchSnapshot()
  })

  it("renders an active tab", () => {
    const tabs = [{ label: "One", active: true }, { label: "Two" }]
    const { container, getByText } = render(<Tabs tabs={tabs} />)

    expect(container.firstChild).toMatchSnapshot()
    expect(getByText("One").classList.contains(styles.rowTabActive)).toBe(true)
  })

  it("renders a disabled tab", () => {
    const tabs = [{ label: "One", disabled: true }, { label: "Two" }]
    const { container, getByText } = render(<Tabs tabs={tabs} />)

    expect(container.firstChild).toMatchSnapshot()
    expect(getByText("One").classList.contains(styles.rowTabDisabled)).toBe(
      true
    )
  })

  it("renders a tab with an onClick", () => {
    const onClick = jest.fn()
    const tabs = [{ label: "One", onClick }, { label: "Two" }]
    const { container, getByText } = render(<Tabs tabs={tabs} />)

    expect(container.firstChild).toMatchSnapshot()

    fireEvent.click(getByText("One"))
    expect(onClick).toHaveBeenCalled()
  })

  it("renders a tab with a href", () => {
    const href = "//example"
    const tabs = [{ label: "One", href }, { label: "Two" }]
    const { container, getByText } = render(<Tabs tabs={tabs} />)

    expect(container.firstChild).toMatchSnapshot()
    expect(getByText("One").getAttribute("href")).toBe(href)
  })

  describe("when using a custom tab renderer", () => {
    it("passes in the right arguments to the a custom tab renderer", () => {
      const tabs = [{ label: "One" }]
      const { baseElement } = render(
        <Tabs
          tabs={tabs}
          renderTab={({
            // tslint:disable-next-line: no-shadowed-variable
            tab,
            tabClassName,
            activeTabClassName,
            disabledTabClassName,
          }) => {
            return (
              <div key={tab.label}>
                <span>{tab.label}</span>
                <span>{tabClassName}</span>
                <span>{activeTabClassName}</span>
                <span>{disabledTabClassName}</span>
              </div>
            )
          }}
        />
      )
      const tab = baseElement.firstChild as HTMLDivElement
      const spans = tab.querySelectorAll("span")

      expect(spans[0].innerHTML).toBe("One")
      expect(spans[1].innerHTML).toBe(styles.rowTab)
      expect(spans[2].innerHTML).toBe(styles.rowTabActive)
      expect(spans[3].innerHTML).toBe(styles.rowTabDisabled)
    })
  })
})
