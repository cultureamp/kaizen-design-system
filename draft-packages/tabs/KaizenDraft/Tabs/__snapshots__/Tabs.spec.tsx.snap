// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tabs renders a disabled tab 1`] = `
<div
  class="container"
>
  <a
    class="disabledTab"
  >
    One
  </a>
  <a
    class="tab"
  >
    Two
  </a>
</div>
`;

exports[`Tabs renders a tab with a href 1`] = `
<div
  class="container"
>
  <a
    class="tab"
    href="//example"
  >
    One
  </a>
  <a
    class="tab"
  >
    Two
  </a>
</div>
`;

exports[`Tabs renders a tab with an onClick 1`] = `
<div
  class="container"
>
  <a
    class="tab"
  >
    One
  </a>
  <a
    class="tab"
  >
    Two
  </a>
</div>
`;

exports[`Tabs renders an active tab 1`] = `
<div
  class="container"
>
  <a
    class="activeTab"
  >
    One
  </a>
  <a
    class="tab"
  >
    Two
  </a>
</div>
`;

exports[`Tabs renders basic tabs 1`] = `
<div
  class="container"
>
  <a
    class="tab"
  >
    One
  </a>
  <a
    class="tab"
  >
    Two
  </a>
</div>
`;
